src/connection/query.ts

```diff
@@ -96,6 +98,15 @@ export class Query<T> {
       throw new Error(queryResult.error || `Failed to run query (unknown error)`);
     }
     this.correlationId = queryResult.id;
+    
+    if (this.sql.toUpperCase().startsWith("CREATE")) {

```

Comment:
This logic doesn't belong in this class. Perhaps it would be best suited in the `vscode-db2i.runEditorStatement` command as defined in `src/views/results/index.ts`, after we've parsed the statement and got `StatementInfo`. Perhaps a property could be added to return the references in `parseStatement`

Reply:
The logic behind this is we only want to signal a change in a schema if we see a create statement, AND the operation was successful. Is there a way to detect if an SQL statement is valid before running? 

For instance, if I copy the same logic here to `src/views/results/index.ts`, I will add the current schema to the changedCache even if my create statement didnt work, thus the next time we query that schema, we will not use the cached items.